[AU4] Adjust track panel header according to style spec

kryksyh opened the pull request:
Resolves: [#6747](https://github.com/audacity/audacity/issues/6747)

*(short description of the changes and the motivation to make the changes)*

<!-- Use "x" to fill the checkboxes below like [x] -->

- [x] I signed [CLA](https://www.audacityteam.org/cla/)
- [x] The title of the pull request describes an issue it addresses
- [x] If changes are extensive, then there is a sequence of easily reviewable commits
- [x] Each commit's message describes its purpose and effects
- [x] There are no behavior changes unnecessary for the stated purpose of the PR

Recommended:
- [x] Each commit compiles and runs on my machine without known undesirable changes of behavior

Eism commented on au4/src/projectscene/qml/Audacity/ProjectScene/trackspanel/TrackSelectionBar.qml:
```diff
@@ -0,0 +1,11 @@
+Rectangle {
+    Layout.fillHeight: true
+
+    property bool isSelected: false
```
properties should be higher than local properties value settings

Eism commented on au4/src/projectscene/qml/Audacity/ProjectScene/trackspanel/TrackItem.qml:
```diff
@@ -157,11 +163,9 @@ ListItemBlank {
-
-                // iconCode
-
+                Layout.preferredHeight: 24
```
The code becomes more readable if separate properties value settings ​​into groups using empty lines

kryksyh commented on au4/src/projectscene/internal/projectviewstate.cpp:
```diff
@@ -5,8 +5,8 @@
 
-constexpr int DEFAULT_HEIGHT = 114;
-constexpr int MIN_HEIGHT = 52;
+constexpr int DEFAULT_HEIGHT = 116;
```
@grliszas14 Actually, it does make more sense to keep the separator as part of the "track item" as it should be drawn after the last track also. So, I'll keep the default height as 116.

grliszas14 replied:
Then maybe talk it through with Alex - this property is shared between track header and clip which height is 114px by design. It's just two pixels but I guess we want to have very fine-tuned UI ;)

grliszas14 replied:
ps clips and separators are obviously separated so that make: 1px of upper separator + 114px clip + 1px separator which is exactly 116px -> we need to make sure that track headers and tracks itself match in the UI

kryksyh replied:
@grliszas14 oh, i missed that the same constant was used by the clip drawing. There one issue with it though, clips are separated by 3 pixels now
![image](https://github.com/user-attachments/assets/1c3cd6cf-29ed-455b-b86e-f5dfd1510603)

grliszas14 replied:
Black 1px border belongs to the clip (it's inner border) so actually what we are left with is 1px white separator with 10% opacity. But yeah - you can make sure we match track header precisely anyway, I'm not that sure right now

kryksyh replied:
But, I've got your point, the DEFAULT_SIZE constant does not take into account the separator. I would rather call it the "bottom padding", does it make sense?

At least the current designs have this 2px line below the last track

![image](https://github.com/user-attachments/assets/cdb4138d-e904-4e15-9a2f-bd81da08dcb2)

grliszas14 replied:
I think I discussed that with Alex someday and we decided to leave tracks separators with 1px width but maybe let's make sure.
@DilsonsPickles we need your expertise

DilsonsPickles replied:
In the designs, both track headers AND tracks are 116px with a 2px inset stroke that acts as a separator. The reason I designed it like this was purely to make my life easier when making mock ups. Instead of faffing around with an additional separator component for both the track headers panel and tracks I could just incorporate the separator into the component. 

With that said, I don't mind if the track headers and tracks are 114px with a 2px separator component underneath each header. All I care about is there being a 2px gap between each track header and a 2px gap between each track in the canvas. 

Hope this helps, will leave implementation decision in your capable hands.

kryksyh commented:
@grliszas14 @Eism I've updated the PR addressing your comments and made sure that clips view and track header are in sync

![image](https://github.com/user-attachments/assets/fce9e5e4-83fe-4816-bc62-618d37ef2b55)
